.specify/memory/constitution.md

```diff
@@ -0,0 +1,361 @@
+- **Integration tests**: Test plugin interactions, API integrations, and complex component interactions.
+- **E2E tests**: Use Playwright to test full user workflows across the dashboard UI.
+
+**For IaC code (resource-types-contrib repo)**:
+
+- **Schema validation**: Test YAML schemas validate correctly against expected inputs.
+- **Recipe deployment tests**: Test Bicep/Terraform recipes deploy successfully in test environments.
+- **Integration tests**: Verify recipes integrate correctly with Radius control plane.
+
+**For documentation (docs repo)**:
+
+- **Build verification**: Hugo builds complete without errors.
+- **Link validation**: All internal and external links resolve correctly.
+- **Spelling validation**: Use `pyspelling` to catch typos and maintain consistency.
+- **Example validation**: All code examples build and run successfully.
+
+Tests MUST be written during feature implementation, not as an afterthought. Code coverage reports are reviewed in PRs to ensure adequate test coverage. New features MUST NOT be merged without corresponding tests at appropriate pyramid levels. Tests MUST fail before implementation (Red-Green-Refactor cycle).
+
+**Rationale**: Quality and reliability are paramount for a platform managing critical infrastructure across multiple clouds. Repository-specific testing ensures bugs are caught early while respecting the unique validation needs of Go services, UI components, infrastructure recipes, and documentation.
+
+### V. Collaboration-Centric Design
+
+Design specifications MUST explicitly address how the feature enables collaboration between developers and platform engineers. Features MUST consider both perspectives:
+
+- **Developer experience**: How does this simplify application authoring and deployment? Does it reduce cognitive load?
+- **Platform engineer experience**: How does this enable governance, compliance, and operational best practices?
+
+Environment and Recipe abstractions MUST be designed to allow platform engineers to define infrastructure patterns while giving developers the flexibility they need. APIs and tooling MUST support both audiences without forcing one perspective onto the other.
+
+**Rationale**: Radius exists to bridge the gap between development and operations teams. Every feature should reinforce this collaboration rather than creating new silos or imposing unnecessary constraints.
+
+### VI. Infrastructure as Code Integration
+
+All infrastructure-related features MUST support Bicep as the primary authoring experience. Bicep type definitions in `bicep-types/` MUST be generated from TypeSpec definitions and kept in sync through `make generate`. Features SHOULD integrate with Terraform through the Recipe system where appropriate. Kubernetes manifests MUST be supported for native Kubernetes resources through existing integration patterns in `pkg/kubernetes/`.
```

Comment:
I'm not sure this is the right order from a user perspective. Terraform should be the primary recipe authoring experience since it's more widely-known